code/WorkInProgress/recycling/conveyor.dm

```diff
@@ -764,6 +970,56 @@ ADMIN_INTERACT_PROCS(/obj/machinery/conveyor_switch, proc/trigger)
 				C.setdir()
 		SEND_SIGNAL(src,COMSIG_MECHCOMP_TRANSMIT_SIGNAL,"switchTriggered")
 
+/obj/machinery/conveyor_switch/built/
+	desc = "A conveyor control switch. This one looks like it was built recently."
+
+
+	New()
+		. = ..()
+		UnsubscribeProcess()
+		START_TRACKING
+		UpdateIcon()
+		AddComponent(/datum/component/mechanics_holder)
+		SEND_SIGNAL(src,COMSIG_MECHCOMP_ADD_INPUT,"trigger", PROC_REF(trigger))
+		SEND_SIGNAL(src,COMSIG_MECHCOMP_ADD_CONFIG,"Refresh ID", PROC_REF(get_new_id))
+		src.get_new_id()
+
+	proc/get_new_id()
+		src.id = TIME // Create a new ID on the fly. It doesn't need to be readable, it just needs to be unique.
+		src.conveyors = list()
+
+	MouseDrop_T(dropped, mob/user)
+		if (ispulsingtool(dropped))
+			if (!istype(dropped, /obj/machinery/conveyor_switch))
+				return ..()
+			var/obj/machinery/conveyor_switch/connected_switch = dropped
+			for (var/obj/machinery/conveyor/conveyor in src.conveyors)
+				conveyor.id = connected_switch.id
+				conveyor.linked_switches += connected_switch
+
+			src.id = connected_switch.id
+			connected_switch.conveyors = src.conveyors
+			user.show_text("You connect \the [src] to \the [dropped]. Both share the same ID now.", "blue")
+
+		return ..()
+
+	attackby(var/obj/item/I, mob/user)
+		if (ispulsingtool(I))
+			var/datum/component/mechanics_connector/connector = I.GetComponent(/datum/component/mechanics_connector)
+			if (!connector)
+				return ..()
+			if (!istype(connector.connectee, /obj/machinery/conveyor_switch))
+				return ..()
+
+			var/obj/machinery/conveyor_switch/connected_switch = connector.connectee
+			src.id = connected_switch.id
+			connected_switch.conveyors = src.conveyors
+			for (var/obj/machinery/conveyor/conveyor in src.conveyors)
+				conveyor.id = connected_switch.id
+				conveyor.linked_switches += connected_switch
+			user.show_text("You connect \the [src] to \the [connector.connectee]. Both share the same ID now.", "blue")
+
+		return ..()
```

Comment:
This does not seem to be usable for me, since the mechanics-component-connecting-component triggers before it and then cancels the thing?

It just opens the normal "link devices" popup for mechanics components. Not sure what the best way to solve this would be...

Reply:
The best solution to this is just not doing it at all. I'll include the change in the changelogs.